src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -40,30 +43,87 @@
         </div>
 
         <div class="mt-2">
-            <div class="btn btn-warning btn-sm" id="generatePassword">@T["Generate password"]</div>
+            <div class="btn btn-warning btn-sm" id="generatePassword" data-password-length="@IdentityOptions.Value.Password.RequiredLength">@T["Generate password"]</div>
             <div class="btn btn-info btn-sm" id="copyPassword">@T["Copy password"]</div>
         </div>
     </div>
-
     <script at="Foot" asp-name="userFieldsPassword">
         document.addEventListener('DOMContentLoaded', function() {
-            const strongPasswordLength = 12;
             let password = document.getElementById('@Html.IdFor(p => p.Password)');
             let passwordConfirmation = document.getElementById('@Html.IdFor(p => p.PasswordConfirmation)');
             let passwordToggle = document.getElementById('passwordToggle');
             let passwordConfirmationToggle = document.getElementById('passwordConfirmationToggle');
             let generate = document.getElementById('generatePassword');
             let copy = document.getElementById('copyPassword');
-            function toggetIcons(element) {
+            const requiredPasswordLength = parseInt(generate.getAttribute('data-password-length'));
+            const allowed = {
+                uppers: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+                lowers: 'abcdefghijklmnopqrstuvwxyz',
+                digits: '0123456789',
+                symbols: '!#$%&\()*+,-./:;<=>\'?@@[\]^_`{|}~'
+            };
+            const getRandomChar = (str) => str.charAt(Math.floor(Math.random() * str.length));
+            const shuffle = (str) => str.sort(function() { return 0.5 - Math.random() });
+            const copyPassword = (str) => navigator.clipboard.writeText(str);
+            const meetUniqueRule = (str) => {
+                let uniqueCharacters = str.filter((item, i, ar) => ar.indexOf(item) === i);
+
+                return uniqueCharacters.length >= @(IdentityOptions.Value.Password.RequiredUniqueChars);
+            };
+            const generatePassword = () => {
```

Comment:
Can we move this function to asset file instead of keeping inline arrow function in cshtml?

Reply:
@ns8482e done.